src/spdx/parser/tagvalue/parser.py

```diff
@@ -0,0 +1,546 @@
+                  "pkg_license_declared : PKG_LICENSE_DECLARED license_or_no_assertion_or_none\n "
+                  "pkg_file_name : PKG_FILE_NAME LINE\n "
+                  "pkg_license_concluded : PKG_LICENSE_CONCLUDED license_or_no_assertion_or_none\n "
+                  "package_version : PKG_VERSION LINE\n pkg_license_comment : PKG_LICENSE_COMMENT text_or_line\n "
+                  "snippet_spdx_id : SNIPPET_SPDX_ID LINE\n snippet_name : SNIPPET_NAME LINE\n "
+                  "snippet_comment : SNIPPET_COMMENT text_or_line\n "
+                  "snippet_copyright_text : SNIPPET_COPYRIGHT_TEXT line_or_no_assertion_or_none\n "
+                  "snippet_license_comment : SNIPPET_LICENSE_COMMENT text_or_line\n "
+                  "file_spdx_id : SNIPPET_FILE_SPDXID LINE\n "
+                  "snippet_license_concluded : SNIPPET_LICENSE_CONCLUDED license_or_no_assertion_or_none\n "
+                  "annotation_spdx_id : ANNOTATION_SPDX_ID LINE\n "
+                  "annotation_comment : ANNOTATION_COMMENT text_or_line\n "
+
+                  )
+    def p_generic_value(self, p):
+        if p[1] in EXPECTED_START_TAG_ELEMENT.keys():
+            self.initialize_new_current_element(EXPECTED_START_TAG_ELEMENT[p[1]])
+        if self.check_that_current_element_matches_class_for_value(TAG_DATA_MODEL_FIELD[p[1]][0], p.lineno(1)):
+            set_value(p, self.current_element)
+
+    @grammar_rule("unknown_tag : UNKNOWN_TAG text_or_line\n | UNKNOWN_TAG DATE\n | UNKNOWN_TAG PERSON_VALUE \n"
+                  "| UNKNOWN_TAG")
+    def p_unknown_tag(self, p):
+        self.logger.append(f"Unknown tag provided in line {p.lineno(1)}")
+
+    @grammar_rule("text_or_line : TEXT")
+    def p_text(self, p):
+        p[0] = str_from_text(p[1])
+
+    @grammar_rule("text_or_line : LINE\n line_or_no_assertion : LINE\nline_or_no_assertion_or_none : text_or_line")
+    def p_line(self, p):
+        p[0] = p[1]
+
+    @grammar_rule("license_or_no_assertion_or_none : NO_ASSERTION\n actor_or_no_assertion : NO_ASSERTION\n"
+                  "line_or_no_assertion : NO_ASSERTION\n line_or_no_assertion_or_none : NO_ASSERTION")
+    def p_no_assertion(self, p):
+        p[0] = SpdxNoAssertion()
+
+    @grammar_rule("license_or_no_assertion_or_none : NONE\n line_or_no_assertion_or_none : NONE")
+    def p_none(self, p):
+        p[0] = SpdxNone()
+
+    @grammar_rule("license_or_no_assertion_or_none : LINE")
+    def p_license(self, p):
+        p[0] = get_spdx_licensing().parse(p[1])
+
+    @grammar_rule("actor_or_no_assertion : PERSON_VALUE\n | ORGANIZATION_VALUE")
+    def p_actor_values(self, p):
+        p[0] = ActorParser.parse_actor(p[1])
+
+    @grammar_rule("spdx_id : SPDX_ID LINE")
+    def p_spdx_id(self, p):
+        # We assume that the documents spdx_id is defined first in the SPDXDocument, before any package or file
+        # information. If this is not the case the parser will behave unexpectedly as the spdx_ids are assigned falsy.
```

Comment:
I don't think that really captures the essence of the problem. My suggestion would be:
"As all SPDX Ids share the same tag, there is no knowing which spdx_id belongs to the document. We assume that to be the first spdx_id we encounter. As the specification does not explicitly require this, our approach might lead to unwanted behavior when the document's SPDX Id is defined later in the document."

Reply:
fixed